packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -497,225 +566,153 @@ export default class SegmentationStateManager {
-      delete this.state.viewports[viewportId][segmentationRepresentationUID];
-    });
-  }
-
-  /**
-   * Set the active segmentation representation for the give viewport
-   * @param viewportId - The Id of the tool group that owns the
-   * segmentation data.
-   * @param segmentationRepresentationUID - string
-   */
-  setActiveSegmentationRepresentation(
+  removeSegmentationRepresentation(
     viewportId: string,
-    segmentationRepresentationUID: string
+    segmentationId: string,
+    type: SegmentationRepresentations
   ): void {
-    Object.keys(this.state.viewports[viewportId]).forEach((segRepUID) => {
-      this.state.viewports[viewportId][segRepUID].active = false;
-    });
-
-    if (!this.state.viewports[viewportId]) {
-      this.state.viewports[viewportId] = {};
-    }
-
-    if (!this.state.viewports[viewportId][segmentationRepresentationUID]) {
-      this.state.viewports[viewportId][segmentationRepresentationUID] = {
-        active: false,
-        visible: true,
-        segmentsHidden: new Set(),
-      };
-    }
-
-    this.state.viewports[viewportId][segmentationRepresentationUID].active =
-      true;
-  }
+    const viewport = this.state.viewports[viewportId];
 
-  getActiveSegmentationRepresentation(
-    viewportId: string
-  ): SegmentationRepresentation | undefined {
-    if (!this.state.viewports?.[viewportId]) {
+    if (!viewport) {
       return;
     }
 
-    const activeSegRep = Object.entries(this.state.viewports[viewportId]).find(
-      ([, value]) => value.active
+    const viewportRendering = viewport.find(
+      (segRep) =>
+        segRep.segmentationId === segmentationId && segRep.type === type
     );
 
-    if (!activeSegRep) {
+    if (!viewportRendering) {
       return;
     }
 
-    return this.getSegmentationRepresentation(activeSegRep[0]);
-  }
-
-  /**
-   * Returns the global segmentation representation config.
-   * @returns The global segmentation representation config object.
-   */
-  getGlobalConfig(): GlobalConfig {
-    return this.state.globalConfig;
+    viewport.splice(viewport.indexOf(viewportRendering), 1);
   }
 
-  /**
-   * Sets the global segmentation representation config.
-   * @param config - The global segmentation representation config object to set.
-   */
-  setGlobalConfig(config: GlobalConfig): void {
-    this.state.globalConfig = config;
-  }
-
-  _getRepresentationConfig(segmentationRepresentationUID: string): {
-    allSegments?: RepresentationConfig;
-    perSegment?: RepresentationConfig;
-  } {
-    const segmentationRepresentation = this.getSegmentationRepresentation(
-      segmentationRepresentationUID
-    );
+  setActiveSegmentation(viewportId: string, segmentationId: string): void {
+    const viewport = this.state.viewports[viewportId];
 
-    if (!segmentationRepresentation) {
+    if (!viewport) {
       return;
     }
 
-    return segmentationRepresentation.config;
+    Object.entries(viewport).forEach(([key, value]) => {
```

Comment:
You don't use the key, so you could use Object.values(viewport).forEach(value => ...